homeassistant/components/drop_connect/sensor.py

```diff
@@ -0,0 +1,432 @@
+                DROP_TemperatureSensorF(coordinator),
+            ]
+        )
+    elif device_type == DEV_PUMP_CONTROLLER:
+        entities.extend(
+            [
+                DROP_CurrentFlowRateSensor(coordinator),
+                DROP_CurrentSystemPressureSensor(coordinator),
+                DROP_TemperatureSensorC(coordinator),
+                DROP_TemperatureSensorF(coordinator),
+            ]
+        )
+    elif device_type == DEV_RO_FILTER:
+        entities.extend(
+            [
+                DROP_FilterCart1Sensor(coordinator),
+                DROP_FilterCart2Sensor(coordinator),
+                DROP_FilterCart3Sensor(coordinator),
+                DROP_InletTdsSensor(coordinator),
+                DROP_OutletTdsSensor(coordinator),
+            ]
+        )
+
+    async_add_entities(entities)
+
+
+class DROP_CurrentFlowRateSensor(DROP_Entity, SensorEntity):
+    """Monitors the current water flow rate."""
+
+    _attr_icon = FLOW_ICON
+    _attr_native_unit_of_measurement = "gpm"
+    _attr_state_class: SensorStateClass = SensorStateClass.MEASUREMENT
+    _attr_translation_key = "current_flow_rate"
+
+    def __init__(self, device) -> None:
+        """Initialize the current flow rate sensor."""
+        super().__init__(self._attr_translation_key, device)
+
+    @property
+    def native_value(self) -> float | None:
+        """Return the current flow rate."""
+        if self._device.current_flow_rate is None:
+            return None
+        return round(self._device.current_flow_rate, 1)
+
+
+class DROP_PeakFlowRateSensor(DROP_Entity, SensorEntity):
+    """Monitors the peak water flow rate for the day."""
+
+    _attr_icon = FLOW_ICON
+    _attr_native_unit_of_measurement = "gpm"
+    _attr_state_class: SensorStateClass = SensorStateClass.MEASUREMENT
+    _attr_translation_key = "peak_flow_rate"
+
+    def __init__(self, device) -> None:
+        """Initialize the peak flow rate sensor."""
+        super().__init__(self._attr_translation_key, device)
+
+    @property
+    def native_value(self) -> float | None:
+        """Return the peak flow rate."""
+        if self._device.peak_flow_rate is None:
+            return None
+        return round(self._device.peak_flow_rate, 1)
+
+
+class DROP_WaterUsedTodaySensor(DROP_Entity, SensorEntity):
+    """Monitors the total water used for the day."""
+
+    _attr_device_class = SensorDeviceClass.WATER
+    _attr_native_unit_of_measurement = UnitOfVolume.GALLONS
+    _attr_translation_key = "water_used_today"
+
+    def __init__(self, device) -> None:
+        """Initialize the water used today sensor."""
+        super().__init__(self._attr_translation_key, device)
+
+    @property
+    def native_value(self) -> float | None:
+        """Return the total used today."""
+        if self._device.water_used_today is None:
+            return None
+        return round(self._device.water_used_today, 1)
+
+
+class DROP_AverageWaterUsedSensor(DROP_Entity, SensorEntity):
```

Comment:
I think you should take a look at entity descriptions. It's a quick way to create more sensors with the same logic. For a simple example checkout the `youtube` integration.

Reply:
And since you are creating different sensors on different devices, checkout `MEASUREMENT_SENSORS` in `withings`, this way you can still refer to separate entity descriptions

Reply:
Thanks for this advice. I've refactored the sensors to use SensorEntityDescription in de87e4e1621315757f6c4225677cf5fed7a486ee.

Reply:
This is starting to look a lot better :)